src/main/java/seedu/address/logic/commands/AddLogCommand.java

```diff
@@ -0,0 +1,194 @@
+        Name name = personToEdit.getName();
+        Phone phone = personToEdit.getPhone();
+        Email email = personToEdit.getEmail();
+        Address address = personToEdit.getAddress();
+        Set<Tag> tags = personToEdit.getTags();
+        List<Log> updatedLogs = addLogDescriptor.getLogs(personToEdit); // main logic encompassed here
+        return new Person(name, phone, email, address, tags, updatedLogs);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof AddLogCommand)) {
+            return false;
+        }
+
+        // cast
+        AddLogCommand a = (AddLogCommand) other;
+        return this.index.equals(a.index)
+                && this.addLogDescriptor.equals(a.addLogDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return "Index: " + this.index.toString() + "\nContent:\n" + this.addLogDescriptor.toString();
+    }
+
+    /**
+     * Stores the details of the edited log to edit a person's logs with, as well as the person's
+     * original details.
+     */
+    public static class AddLogDescriptor {
+
+        private String newTitle;
+        private String newDescription;
+
+        /**
+         * Constructs a new {@code AddLogDescriptor} object.
+         */
+        public AddLogDescriptor() {
+            this.newTitle = null;
+            this.newDescription = null;
+        }
+
+        public void setNewTitle(String newTitle) {
+            this.newTitle = newTitle;
+        }
+
+        public void setNewDescription(String newDescription) {
+            this.newDescription = newDescription;
+        }
+
+        /**
+         * Returns true if title has been edited.
+         */
+        public boolean isTitleEdited() {
+            return this.newTitle != null;
+        }
+
+        /**
+         * Returns a list of {@code Log} objects that include the {@code Person}'s original logs
+         * as well as the new logs.
+         */
+        public List<Log> getLogs(Person personToEdit) throws CommandException {
+            try {
+                Log toAdd = new Log(this.newTitle, this.newDescription); // create log to be added
+                if (personToEdit.containsLog(toAdd)) {
+                    throw new CommandException(MESSAGE_DUPLICATE_LOG); // ensure not a duplicate log being inserted
+                }
+                List<Log> newLogs = new ArrayList<Log>(personToEdit.getLogs());
+                newLogs.add(toAdd); // add log
+                return newLogs;
+
+            } catch (IllegalArgumentException ie) {
+                throw new CommandException(Log.TITLE_CONSTRAINTS); // illegal title
+            } catch (NullPointerException ne) {
+                throw new CommandException(AddLogCommandParser.MESSAGE_INVALID_FORMAT);
```

Comment:
Is there any reason you handle the exception of Log constructor and AddLogCommandParser here? Maybe we should shift the exception handler for the above two classes into the respective classes themselves? I feel that it is better practice as it encapsulates all the logic within the relevant classes.

Reply:
Hm, I'm not sure about this actually. I agree that putting them in their respective class alone (they are already in those classes, by the way) would be cleaner, but the rationale for including them here was also to ensure multiple layers of checks. You can see the relevant test files in `AddLogCommandTest` to see what I mean. i.e. suppose the parser somehow lets something through, this will be an added layer of checks. 

What do @dionegoh and @limweiliang think?

Reply:
I am leaning towards handling the exceptions in their respective classes to prevent duplicate checks whenever possible and also for the purpose of better encapsulation!